pallets/pools/src/tranche.rs

```diff
@@ -1453,6 +1490,41 @@ where
 	Ok(risk_buffers)
 }
 
+pub(crate) fn create_asset_metadata<Balance, CurrencyId>(
+	decimals: u32,
+	currency: CurrencyId,
+	parachain_id: ParaId,
+	token_name: Vec<u8>,
+	token_symbol: Vec<u8>,
+) -> Result<AssetMetadata<Balance, CustomMetadata>, DispatchError>
+where
+	Balance: Zero,
+	CurrencyId: Encode,
+	CustomMetadata: Parameter + Member + TypeInfo,
+{
+	// Is this the right ID?
+	let tranche_id = currency.encode();
```

Comment:
Is this the right `tranche_id` @branan @mustermeiszer ?

Reply:
I think it is correct, as the `CurrencyId::Tranche(PoolId, TrancheId)` is a unique Id in our system, while `tranche_id`'s are only unique per pool. 

cc: @NunoAlexandre for clarity, can you confirm that the `GeneralKey` we are using the id for should have this structure and properties?